### PR TITLE
Feat/navigation

### PR DESCRIPTION
**Pull request by urchykli**

### CODE REVIEW

File: wp-content/themes/csisnuclearnetwork/inc/site-nav-walker.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        $id = apply_filters( 'nav_menu_item_id', 'menu-item-'. $item->ID, $item, $args );
+        $id = $id ? ' id="' . esc_attr( $id ) . '"' : '';
+        $output .= $indent . '<li' . $id . $class_names . '>';
+        $attributes  = ! empty( $item->attr_title ) ? ' title="'  . esc_attr( $item->attr_title ) .'"' : '';
+        $attributes .= ! empty( $item->target )     ? ' target="' . esc_attr( $item->target     ) .'"' : '';
+        $attributes .= ! empty( $item->xfn )        ? ' rel="'    . esc_attr( $item->xfn        ) .'"' : '';
+        $attributes .= ! empty( $item->url )        ? ' href="'   . esc_attr( $item->url        ) .'"' : '';
+        $item_output = $args->before;
+        if ( $args->walker->has_children && $depth == 0 ) {
+          $item_output .= '<button>' . $attributes;
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/inc/site-nav-walker.php:92]: I would recommend removing the `$attributes` from the instances where you're using `<button>`. If you look at what the attributes are (defined on lines 86-89), they're pretty link specific, so we can remove them from here.

File: wp-content/themes/csisnuclearnetwork/inc/site-nav-walker.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+        $class_names = $value = '';
+        $classes = empty( $item->classes ) ? array() : (array) $item->classes;
+        $classes[] = 'sub-menu sub-menu-depth-' . $depth;
+        $class_names = join( ' ', apply_filters( 'nav_menu_css_class', array_filter( $classes ), $item, $args ) );
+        $class_names = $class_names ? ' class="' . esc_attr( $class_names ) . '"' : '';
+
+        $indent = str_repeat("\t", $depth);
+        $output .= "\n$indent<div class='sub-menu-container sub-menu-container-depth-" . $depth . " row'>
+            <div class='col-xs-12 col-md sub-menu-children' aria-expanded='false'>\n";
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/inc/site-nav-walker.php:48]: Whenever you see classes like `col-xs-12` or `col-md`, you can remove those. The ASP site uses a Bootstrap derivative stylesheet based on flexbox that has a 12 column layout and that's what those classes are from. NN doesn't use that, so go ahead and remove it.

File: wp-content/themes/csisnuclearnetwork/assets/_scss/layout/_header.scss [hunk 3]
```diff
@@ -327,31 +342,54 @@
   &__menu {
```
Comment (jnschrag) [inline wp-content/themes/csisnuclearnetwork/assets/_scss/layout/_header.scss:342]: I would maybe recommend moving the navigation styles into their own file and keeping the `_header.scss` file more layout + logo stuff, but up to you.

Don't forget to add a cursor to the buttons/links in the menu - the browser won't add that automatically to buttons, so you'll need to do it.

**Review by jnschrag — CHANGES_REQUESTED** [review]
Really, really nice work here, Lindsay! I'm super proud of what you accomplished with this. This was not an easy feat but it looks great. :)

**Comment by urchykli** [comment]
This PR handles the navigation menu. It does not handle the rest of the site header including the scroll functionality and the search bar.

The submenus are designed to appear and disappear on click or focus rather than hover.

**Review by jnschrag — CHANGES_REQUESTED** [review]
This is looking really good, Lindsay. A couple of tweaks and I think this is golden. Some of these could definitely be nice to haves too. :)

- [x] The spacing between "All Analysis" and "All Series" is off. This is the CSS Grid trying to align the items. If you change the "Popular Topics" so it spans 3 rows instead of 2, it'll fix itself.
- [ ] The dropdown is missing a carrot/triangle at the top on larger screens
- [ ] Check the hover color of the post title of the featured post
- [ ] I'm also not seeing a change in the hover state on the links in the drop down menus
- [ ] On mobile we're missing an overlay over the content when the menu is open. A common request is to "freeze" the body so the user can't scroll on anything but the menu when the menu is open. I like the [body-scroll-lock](https://github.com/willmcpo/body-scroll-lock) library for that. You would add the `disableBodyScroll` to the same event listeners that open & close the menu.

**Pull request merged**